### PR TITLE
feat: add feed component

### PR DESCRIPTION
**Pull request by OsgoodPaekid**
#14

### CODE REVIEW

**Comment by OsgoodPaekid** [comment]
Added the content of a tweet: the like, retweet, comment, share and tweet impressions icons.

**Comment by OsgoodPaekid** [comment]
Changed the FeedItem component to a reusable component and rendered. And I also converted the png icon to an svg

**Comment by OsgoodPaekid** [comment]
Used counts instead of components and moved all components to the FeedItem component and assigned props.

**Comment by OsgoodPaekid** [comment]
created a formateDate function that displays time/date since tweet was created and properly aligned the icons and text in the tweet
WhatsApp Image 2023-07-08 at 22 04 08

File: src/components/feed/index.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            <div className="flex">
+                    <div>
+                        <img src={props.avatar} className="h-16 rounded-full ml-4 mr-4"/>
+                    </div>
+                    <div>
+                        <div>
+                            <div className="flex ">
+                                <div className="font-bold pr-2">{props.username}</div>
+                                <div>{props.handle}</div>
+                                <span className="pl-2">{props.time}<span>h</span></span>
```
[the remark this answers is not in the document]
Comment (OsgoodPaekid) [reply src/components/feed/index.js]: resolved

File: src/app/page.js [hunk 2]
```diff
@@ -1,5 +1,32 @@
...
 export default function Home() {
-  return <div>Feed</div>;
+
+      const items = [{
+        id: 1,
+        avatar: "https://ionicframework.com/docs/img/demos/avatar.svg",
+        username: "Osgood",
+        handle: "@osgoodpaekid",
+        // time: new Date().toISOString().slice(11, 19),
+        time: new Date().getHours(),
```
[the remark this answers is not in the document]
Comment (OsgoodPaekid) [reply src/app/page.js]: resolved

File: src/components/feed/index.js [hunk 3]
```diff
@@ -6,32 +6,61 @@ import {LuShare} from "react-icons/lu"
 
 
 export function FeedItem(props){
-   
-    return(
-        <div>
-            <div className="flex">
+  
+  function formatDate(){
```
Comment (alphaofficial) [inline src/components/feed/index.js:10]: you can move this function into a `utilities` directory and export from there

File: src/components/feed/index.js [hunk 4]
```diff
@@ -6,32 +6,61 @@ import {LuShare} from "react-icons/lu"
...
+    
+    const past = new Date('2023-07-07');
```
Comment (alphaofficial) [inline src/components/feed/index.js:12]: why is the date hard-coded?
ideally, your component will have the date in there, which you can pass to your function.
tip: read about pure functions

File: src/app/page.js [hunk 5]
```diff
@@ -9,7 +9,7 @@ export default function Home() {
         username: "Osgood",
         handle: "@osgoodpaekid",
         // time: new Date().toISOString().slice(11, 19),
-        time: new Date().getHours(),
+        // createdAt: new Date().getHours(),
```
Comment (alphaofficial) [inline src/app/page.js:12]: why is it commented out?

**Review by alphaofficial — CHANGES_REQUESTED** [review]
left questions and comments

**Comment by OsgoodPaekid** [comment]
added utilities directory and exported the formatDate function to the feed component

File: src/components/feed/index.js (hunk 3 above)
Comment (OsgoodPaekid) [reply src/components/feed/index.js]: Resolved

File: src/components/feed/index.js (hunk 4 above)
Comment (OsgoodPaekid) [reply src/components/feed/index.js]: Resolved

File: src/app/page.js (hunk 5 above)
Comment (OsgoodPaekid) [reply src/app/page.js]: Resolved

File: src/app/page.js [hunk 6]
```diff
@@ -1,5 +1,31 @@
...
 
 export default function Home() {
-  return <div>Feed</div>;
+
+      const items = [{
+        id: 1,
+        avatar: "https://ionicframework.com/docs/img/demos/avatar.svg",
+        username: "Osgood",
+        handle: "@osgoodpaekid",
+        createdAt: new Date('2023-07-07'),
```
Comment (alphaofficial) [inline src/app/page.js:11]: as mentioned before, this is a date object. You should use iso string

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString

File: src/components/feed/index.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
+import {FaRegComment } from "react-icons/fa"
+import {AiOutlineRetweet} from "react-icons/ai"
+import {MdFavoriteBorder} from "react-icons/md"
+import {VscGraph} from "react-icons/vsc"
+import {LuShare} from "react-icons/lu"
+import { formatDate } from "../../../utilities/formateDate"
```
Comment (alphaofficial) [inline src/components/feed/index.js:6]: alias paths is configured, update this to use alias path. it's cleaner that way

File: utilities/formateDate.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
+export function formatDate(x){
+    
+    const past = new Date(x);
+    const now = new Date();
+
+    const result = now - past;
+    const seconds = Math.floor(result / 1000);
```
Comment (alphaofficial) [inline utilities/formateDate.js:7]: the date object has methods to get seconds, minutes, hours and days. use those instead

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getDay

**Comment by OsgoodPaekid** [comment]
used alias path to make the code more cleaner and also changed the argument for formatDate function for a better understanding of the function

File: src/components/feed/index.js (hunk 7 above)
Comment (OsgoodPaekid) [reply src/components/feed/index.js]: Resolved

File: utilities/formateDate.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
+export function formatDate(x){
```
[the remark this answers is not in the document]
Comment (OsgoodPaekid) [reply utilities/formateDate.js]: Resolved

**Comment by OsgoodPaekid** [comment]
added a condition to check when tweet was created and if it's been more than a week(7 days), it only displays the date the tweet was created

**Comment by OsgoodPaekid** [comment]
without the equal sign, the seventh day was exempted.

**Review by alphaofficial — APPROVED** [review]
LGTM

**Pull request merged**